liquidhaskell-boot/src/Language/Haskell/Liquid/GHC/Plugin.hs

```diff
@@ -196,6 +227,38 @@ unoptimiseDynFlags df = updOptLevel 0 df
     , ghcMode      = CompManager
     }
 
+--------------------------------------------------------------------------------
+-- | Parsing phase -------------------------------------------------------------
+--------------------------------------------------------------------------------
+
+-- | We hook at this stage of the pipeline to store the parsed module
+-- for later processing during typechecking:
+--
+-- * Comments are preserved during parsing because we turn on
+--   'Opt_KeepRawTokenStream'. So we can get the /LH/ specs from there.
+--
+-- * The typechecker hook needs to do desugaring itself (see the
+--   comment on 'typecheckHook') which requires access to a full
+--   'TypecheckedModule'. Annoyingly, the typechecker hook is given
+--   only a 'TcGblEnv', so we have to redo typechecking there. And
+--   the input to that needs to be a 'ParsedModule'.
+--
+parsedHook :: [CommandLineOption] -> ModSummary -> ParsedResult -> Hsc ParsedResult
+parsedHook _ ms parsedResult = do
+  liftIO $ atomicModifyIORef' breadcrumbs $ \breadcrumbs ->
+      let breadcrumbs' = HM.alter (<|> Just (Parsed parsed)) thisModule breadcrumbs
+      in (breadcrumbs', ())
```

Comment:
```suggestion
  _ <- swapBreadcrumbs thisModule (Just (Parsed parsed))
```
?

Reply:
Is that what we want, though? Note that here, we only replace the breadcrumb with `Parsed parsed` if there's no other breadcrumb already.

Reply:
If we should expect to encounter breadcrumbs already inserted for a module, it would deserve a comment to explain how could that happen and what the code is expected to do in such a case.

When I propose the reuse of `swapBreadcrumbs` I was not aware of the possibility.